### PR TITLE
Changes to the lore of Knock

### PR DESCRIPTION
**Pull request by Profakos**
## About The Pull Request

This PR renames Knock to Lock, and changes most of the knowledge gain lore.

## Why It's Good For The Game

The Knock Lore, is based on the Knock Principle from Cultist Simulator, with the path description being copied from the wiki. Many other keywords and concepts are fully lifted from that game (Locksmith's Secret, Mother Of Ants, etc). In my vision, if a heretic path has to be based on a principle from cultist simulator, it should have its own spin, and also, the knowledge gain texts should tell a story. For example, Ash tells the story of a watchman burning down their city after being betrayed, and Cosmic is a love story between a knowledge seeker and a monster from the beyond. 

So I have decided to reflavour Knock. I have changed the name to Lock, so at least it would feel similar, just like how Blade is akin to Edge. Many powers also block people or confuse their paths instead of opening new ways, and thus, I feel a path whose name implies that it *both* opens and closes would be more self describing. 

I have changed most of its lore to be about the Locked Labyrinth, where knowledge seekers willingly trap themselves and submit themselves to servitude to find ultimate freedom by progressing through its trials. These are the Stewards, who are basically workers in an infinite and malicious hotel in their dreams. Consider them assistants if you will (this wasn't my intention when I wrote the lore, but thinking about it in retrospect, it honestly fits). In the implied story, the heretic joins their ranks, but keeps getting closer to the more corrupt members, along with parasitic spirits. Ultimately, they manage to open the Labyrinth's core, letting out the Stewards, allowing them to manifest in the forms of heretic summon creatures.

The side path spells and the lock knowledge ritual I have not touched, they were fine. Some items have been renamed and repathed.

I have kept the distinctive sound effect for using the Grasp, as its unique enough. Though if someone did have a nice sound effect for turning a lock and added some filters, I would add it.

**DB Issue**

I have renamed the achievement's define to MEDAL_LOCK_ASCENSION but kept the value as "Knock", as I don't know how trigger a change in the DB. If this is a blocking change, I'll try to figure out how to make a migration file.

**Future improvements**

I would also come back later with another PR, that hands out names to the eldritch beings spawned by the portal, based on the Stewards in the knowledge gain lore that I added, along with some new ones that fit the theme, and some jokey ones like Minotaur.

## Changelog

:cl:
spellcheck: Renamed Knock to Locks, and changed most of the flavor text of knowledge gain, and renamed some items and knowledges from the path.
/:cl:

### CODE REVIEW

**Comment by Profakos** [comment]
I would like the Sprites tag to be removed, all I did was that I have updated the name of the icon states.

**Comment by mc-oofert** [comment]
1984 you may not copy like 3 things from the cultistsim wiki without the entire thing being rewritten

**Comment by Annihilit** [comment]
Cosmic has nothing to do with Cultist sim so I think we should let Knock rip it off directly.

**Comment by Dorsisdwarf** [comment]
Not having the path be called the same thing as a powerful and common wizard spell will also finally resolve my perpetual confusion

**Comment by Annihilit** [comment]
> Cosmic having nothing to do with Cultist Sim is not relevant to this PR. Its also one of its best qualities.

incredibly incorrect opinion.

**Review by Jacquerel — CHANGES_REQUESTED** [review]
One important note here, and some suggestions.

File: code/datums/achievements/misc_achievements.dm [hunk 1]
```diff
@@ -135,11 +135,11 @@
...
 	icon = "cosmicascend"
 
-/datum/award/achievement/misc/knock_ascension
-	name = "Secrets behind the Spider Door"
+/datum/award/achievement/misc/lock_ascension
+	name = "Secrets of the Locked Labyrinth"
 	desc = "You managed to open a gate into the mansus."
-	database_id = MEDAL_KNOCK_ASCENSION
-	icon = "knockascend"
+	database_id = MEDAL_LOCK_ASCENSION
```
[the remark this answers is not in the document]
Comment (Profakos) [reply code/datums/achievements/misc_achievements.dm:141]: I didn't change the value itself, it still say Knock, but if you think this is safer, I will undo it.

[the remark this answers is not in the document]
Comment (Jacquerel) [reply code/datums/achievements/misc_achievements.dm:141]: As long as the value is the same that is fine then

[the remark this answers is not in the document]
Comment (Profakos) [reply code/datums/achievements/misc_achievements.dm:141]: image
Double checked, the value is indeed the same.

**Comment by JupiterJaeden** [comment]
This is such a cool PR for real

**Pull request merged**